crates/oxc_ast/src/ast/js.rs

```diff
@@ -1635,6 +1639,12 @@ pub struct MethodDefinition<'a> {
     pub span: Span,
     pub decorators: Vec<'a, Decorator<'a>>,
     pub key: PropertyKey<'a>,
+    #[visit_args(flags = Some(match self.kind {
```

Comment:
@overlookmotel my proposed solution to resolve the flags or other params, `visit_args` would be accompanied by a `#[visited_node(params = (flags: ScopeFlags)]` which defines the extra parameters.